src/finch/tensor.py

```diff
@@ -412,11 +416,36 @@ def random(shape, density=0.01, random_state=None):
     return Tensor(jl.fsprand(*args))
 
 
+def asarray(obj, /, *, dtype=None, format=None):
+    if format not in {"coo", "csr", "csc", "csf", "dense", None}:
+        raise ValueError(f"{format} format not supported.")
+    tensor = obj if isinstance(obj, Tensor) else Tensor(obj)
+
+    if format is not None:
+        order = tensor.get_order()
+        if format == "coo":
+            storage = Storage(SparseCOO(tensor.ndim, Element(tensor.fill_value)), order)
```

Comment:
Will `Tensor.ndim` work for this one?

Reply:
Here we need to provide dimensionality of the input tensor. `ndim` is an instance attribute, not class attribute, so `Tensor.ndim` won't work here.

Reply:
I mean that would NDim work for a COO-format tensor?

Reply:
Resolved in Slack; `ndim` would do the right thing.